.werft/platform-delete-preview-environments-cron.ts

```diff
@@ -422,11 +422,44 @@ async function removePreviewEnvironment(previewEnvironment: PreviewEnvironment)
     }
 }
 
-async function removeCertificate(preview: string, kubectlConfig: string, slice: string) {
-    return exec(
-        `kubectl --kubeconfig ${kubectlConfig} -n certs delete --ignore-not-found=true cert harvester-${preview} ${preview}`,
-        { slice: slice, async: true },
-    );
+async function removeOrphanCertificates() {
+    const certificatesNamespace = "certs"
+    werft.phase(SLICES.DELETING_ORPHAN_CERTIFICATES);
+
+    const certificates = exec(
+        `kubectl --kubeconfig ${CORE_DEV_KUBECONFIG_PATH} get certificates -n ${certificatesNamespace} -o=custom-columns=:metadata.name | grep harvester-`,
+        { slice: SLICES.DELETING_ORPHAN_CERTIFICATES, silent: true, async: false },
+    )
+    .stdout.trim()
+    .split("\n");
+
+    const previews = exec(
+        `kubectl --kubeconfig ${HARVESTER_KUBECONFIG_PATH} get ns -o=custom-columns=:metadata.name | grep preview-`,
+        { slice: SLICES.DELETING_ORPHAN_CERTIFICATES, silent: true, async: false },
+    )
+    .stdout.trim()
+    .replace(/preview-/g, "")
+    .split("\n");
+
+    certificates.forEach(certificate => {
+        const owner = exec(
+            `kubectl --kubeconfig ${CORE_DEV_KUBECONFIG_PATH} get certificates -n ${certificatesNamespace} -o=custom-columns=:metadata.annotations.preview/owner`,
+            { slice: SLICES.DELETING_ORPHAN_CERTIFICATES, silent: true, async: false },
+        ).stdout.trim()
+
```

Comment:
How about ignoring certificates with an empty owner so we can merge this without having to worry about "old" certificates from before we started tracking the owner. Something like:

```ts
if (!owner) {
  werft.log(SLICES.DELETING_ORPHAN_CERTIFICATES, `Certificate ${certificate} does not have an owner. Skipping.`)
  return
}

Reply:
I'm looking at the current previews and certificates now, I believe there's no need to no merge at the moment 🙂. We only have orphan certificates, besides one that the owner preview is almost 2 weeks old (not sure why it wasn't deleted yet)